Sources/StreamChatUI/CommonViews/ListCollectionViewLayout/ListCollectionViewLayout.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// The `ListCollectionViewLayout` delegate to control how to display the list.
+public protocol ListCollectionViewLayoutDelegate: class {
+    /// Implement this method to have detailed control over the visibility of the cell separators.
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: ListCollectionViewLayout,
+        shouldShowSeparatorForCellAtIndexPath indexPath: IndexPath
+    ) -> Bool
+}
+
+/// An `UICollectionViewFlowLayout` implementation to make the collection view behave as a `UITableView`.
+open class ListCollectionViewLayout: UICollectionViewFlowLayout{
+
+    /// The reuse identifier of the cell separator view.
+    open var separatorIdentifier: String = "CellSeparatorIdentifier"
+
+    /// The height of the cell separator view. By default is the hair height, one physical pixel (1 / displayScale).
+    open var separatorHeight: CGFloat {
+        1 / (collectionView?.traitCollection.displayScale ?? 1)
+    }
+
+    weak var delegate: ListCollectionViewLayoutDelegate?
```

Comment:
Let's do it the same way `UICollectionViewDelegateFlowLayout` does it, by accessing `collectionView.delegate`